calico-vpp-agent/watchers/peers_watcher.go

```diff
@@ -283,6 +303,16 @@ func (w *PeerWatcher) resyncAndCreateWatcher(state map[string]*bgpPeer) error {
 				delete(state, ip)
 			}
 		}
+		// Clean up any secrets that are no longer referenced by any bgp peers
+		if w.secretWatcher != nil {
```

Comment:
We should always have `w.secretWatcher != nil` no ?

Reply:
Indeed :)